### PR TITLE
Adding .F90 test for test_target_update_devices.F90

### PR DESCRIPTION
**Pull request by tmh97**
Test passes on spock with cray (ftn version 12.0.1) and on summit with gcc (10.2.0).

Potential bug with xl 16.01
Screen Shot 2021-08-11 at 12 05 22 AM

### CODE REVIEW

**Review by nolanbaker31 — APPROVED** [review]
lgtm

**Comment by spophale** [comment]
Curious, did you try on host only ?

**Review by spophale — COMMENTED** [review]
Check if host fallback works correctly with this test (ideally with LLVM).

**Comment by tmh97** [comment]
@tmh97 check host fallback / no_offloading enabled

File: tests/4.5/target_update/test_target_update_devices.F90 [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+         CALL omp_set_default_device(dev)
+         !$omp target update to(h_matrix)
+         !$omp target map(alloc: h_matrix) map(tofrom: isHost(dev))
+            isHost(dev) = omp_is_initial_device()
+         
+            DO i = 1, N
+               h_matrix(i) = h_matrix(i) + 1
+            END DO
+          !$omp end target
+         !$omp target update from(h_matrix)
```
Comment (spophale) [inline tests/4.5/target_update/test_target_update_devices.F90:81]: overwrite

File: tests/4.5/target_update/test_target_update_devices.F90 [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      END DO
+
+      ! Unmap the matrix
```
Comment (spophale) [inline tests/4.5/target_update/test_target_update_devices.F90:84]: Move unman to after checking

File: tests/4.5/target_update/test_target_update_devices.F90 (hunk 1 above)
Comment (tmh97) [inline tests/4.5/target_update/test_target_update_devices.F90:81]: Move update from to outside this do loop, map sure h_matrix is not getting overwritten, and value of hmatrix is checked on the host for each available device

**Review by spophale — APPROVED** [review]

**Pull request merged**